src/query/server/server.go

```diff
@@ -319,6 +324,38 @@ func newDownsampler(
 	return downsampler
 }
 
+func newDownsamplerAutoMappingRules(
+	namespaces []local.ClusterNamespace,
+	logger *zap.Logger,
+) []downsample.MappingRule {
+	var autoMappingRules []downsample.MappingRule
+	for _, namespace := range namespaces {
+		opts := namespace.Options()
+		attrs := opts.Attributes()
+		if attrs.MetricsType == storage.AggregatedMetricsType {
+			downsampleOpts, err := opts.DownsampleOptions()
+			if err != nil {
+				logger.Fatal("unable to resolve downsample options",
```

Comment:
nit: could you bubble the error up and fatal in main/run

Reply:
Sure thing.